lamindb_setup/core/_hub_core.py

```diff
@@ -83,13 +83,29 @@ def _init_storage(ssettings: StorageSettings, client: Client) -> UUID:
 
 
 def delete_instance_storage(instance_identifier: str) -> None:
+    from .upath import create_path, check_s3_storage_location_empty
+    from ._settings_storage import TOUCH_FILE_PATH
+    from lamindb_setup import settings
+
     owner, name = instance_identifier.split("/")
     instance_account = call_with_fallback_auth(
         select_instance_by_owner_name,
         owner=owner,
         name=name,
     )
+
     if instance_account is not None:
+        # we need to make sure the default 0-byte object in the storage location has
+        # not been deleted to avoid permission errors from leveraging s3fs on an
+        # empty subdirectory
+        path = create_path(
+            instance_account["storage"]["root"], settings.user.access_token
+        )
+        path.fs.touch(str(path / TOUCH_FILE_PATH))
```

Comment:
Please move this into its own function within `_settings_storage.py` and call it something like:
```
def mark_storage_root(root: UPathStr):
     is_initialized_key = ".lamindb/_is_initialized"
     path.fs.touch(str(path / is_initialized_key))
```

And add the comment section there and reuse it in both places?